### PR TITLE
Fix GravityArcMotion path creation in some quadrants

### PR DESCRIPTION
**Pull request by takahirom**
Update ArcMotion by new platform version.
I applied this fix. 
https://android.googlesource.com/platform/frameworks/base/+/eddc8dbed621077ddbc3a66f475920ea5f63283d%5E%21/#F0

And I implement the control points for gravity arc motion. 

output

You can confirm this motion in material-element app.
https://github.com/takahirom/material-element

arc

### CODE REVIEW

File: app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
     private static final float DEFAULT_MIN_ANGLE_DEGREES = 0;
     private static final float DEFAULT_MAX_ANGLE_DEGREES = 70;
     private static final float DEFAULT_MAX_TANGENT = (float)
-            Math.tan(Math.toRadians(DEFAULT_MAX_ANGLE_DEGREES/2));
+            Math.tan(Math.toRadians(DEFAULT_MAX_ANGLE_DEGREES / 2));
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java]: Could you restore previous formatting?

File: app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java [hunk 2]
```diff
@@ -163,23 +150,45 @@ public Path getPath(float startX, float startY, float endX, float endY) {
 
             float minimumArcDist2 = 0;
 
-            if (Math.abs(deltaX) < Math.abs(deltaY)) {
+            if ((Math.abs(deltaX) < Math.abs(deltaY))) {
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java]: There is to many braces. You doubled it. Please fix it.

File: app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java [hunk 3]
```diff
@@ -163,23 +150,45 @@ public Path getPath(float startX, float startY, float endX, float endY) {
...
+                }
 
                 minimumArcDist2 = midDist2 * mMinimumVerticalTangent
                         * mMinimumVerticalTangent;
             } else {
                 // Same as above, but flip X & Y
-                float eDistX = h2 / (2 * deltaX);
-                ex = endX + eDistX;
-                ey = endY;
+                float eDistX = Math.abs(h2 / (2 * deltaX));
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java]: Nice one ;-)

Comment (takahirom) [inline app/src/main/java/io/plaidapp/ui/transitions/GravityArcMotion.java]: Thanks :)

**Comment by takahirom** [comment]
@piotrek1543 I fixed it! :)

**Comment by nickbutcher** [comment]
Thank you for the PR but this is not something we'd like to merge in the project. In the future please feel free to open an issue prior to developing a feature to discuss.

**Pull request closed without merging**